__tests__/index.test.ts

```diff
@@ -31,1073 +50,1197 @@ describe('Some edge cases', () => {
-  checkResult(result, 'mother', f(1,4))
-})
+  test('1 wife, father, mother, 1 full_uncle', () => {
+    const result = calculate({ wife: 1, father: 1, mother: 1, full_paternal_uncle: 1 })
+    checkResult(result, 'wife', f(1,4))
+    checkResult(result, 'father', f(1,2))
+    checkResult(result, 'mother', f(1,4))
+    ensureCompleteResult(result, ['wife', 'father', 'mother'])
+  })
 
-test('husband, 1 paternal_grand_mother, 1 maternal_brother, 1 paternal_cousin', () => {
-  const result = calculate({
-    husband: 1,
-    paternal_grand_mother: 1,
-    maternal_sibling: 1,
-    paternal_cousin: 1
-  })
-  checkResult(result, 'husband', f(1,2))
-  checkResult(result, 'paternal_grand_mother', f(1,6))
-  checkResult(result, 'maternal_sibling', f(1,6))
-  checkResult(result, 'paternal_cousin', f(1,6))
-})
+  test('husband, 1 paternal_grand_mother, 1 maternal_brother, 1 paternal_cousin', () => {
+    const result = calculate({
+      husband: 1,
+      paternal_grand_mother: 1,
+      maternal_sibling: 1,
+      paternal_cousin: 1
+    })
+    checkResult(result, 'husband', f(1,2))
+    checkResult(result, 'paternal_grand_mother', f(1,6))
+    checkResult(result, 'maternal_sibling', f(1,6))
+    checkResult(result, 'paternal_cousin', f(1,6))
+    ensureCompleteResult(result, ['husband', 'paternal_grand_mother', 'maternal_sibling', 'paternal_cousin'])
+  })
 
-xtest('1 wife, father, mother, 2 maternal_brother', () => {
-  const result = calculate({ wife: 1, father: 1, mother: 1, maternal_sibling: 2 })
-  printResults(result)
-  checkResult(result, 'wife', f(1,4))
-  checkResult(result, 'father', f(7,12))
-  checkResult(result, 'mother', f(1,6))
-})
+  xtest('1 wife, father, mother, 2 maternal_brother', () => {
+    const result = calculate({ wife: 1, father: 1, mother: 1, maternal_sibling: 2 })
+    printResults(result)
+    checkResult(result, 'wife', f(1,4))
+    checkResult(result, 'father', f(7,12))
+    checkResult(result, 'mother', f(1,6))
+    ensureCompleteResult(result, ['wife', 'father', 'mother'])
+  })
 
-test('husband, mother, 1 grand_father', () => {
-  const result = calculate({ husband: 1, mother: 1, paternal_grand_father: 1 })
-  checkResult(result, 'husband', f(1,2))
-  checkResult(result, 'mother', f(1,3))
-  checkResult(result, 'paternal_grand_father', f(1,6))
-})
+  test('husband, mother, 1 grand_father', () => {
+    const result = calculate({ husband: 1, mother: 1, paternal_grand_father: 1 })
+    checkResult(result, 'husband', f(1,2))
+    checkResult(result, 'mother', f(1,3))
+    checkResult(result, 'paternal_grand_father', f(1,6))
+    ensureCompleteResult(result, ['husband', 'mother', 'paternal_grand_father'])
+  })
 
-test('husband, mother, 2 paternal_brother, 2 maternal_brother', () => {
-  const result = calculate({ husband: 1, mother: 1, paternal_brother: 2, maternal_sibling: 2 })
-  checkResult(result, 'husband', f(1,2))
-  checkResult(result, 'mother', f(1,6))
-  checkResult(result, 'paternal_brother', f(0,6))
```

Comment:
Same as above.

There might be other instances I missed since this review was done from a phone 🙂